From af42bb31e0a1f856253e0eedd287e3419e2158c4 Mon Sep 17 00:00:00 2001
From: "Yew, Chang Ching" <chang.ching.yew@intel.com>
Date: Wed, 5 Jun 2019 12:28:57 +0800
Subject: [PATCH 1093/1214] Revert "media: intel-ipu4: Generate error signal
 only for faulty stream."

This reverts commit bfb09efcc7e1d38a934a8db35c3e0d6339240b7c.

Change-Id: I645ff74d7a4e3f134f5ed1aa1b7a276f7cf3373c
Tracked-On: PKT-2287
Signed-off-by: Yew, Chang Ching <chang.ching.yew@intel.com>
---
 drivers/media/pci/intel/ipu4/ipu4-isys-csi2.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4-isys-csi2.c b/drivers/media/pci/intel/ipu4/ipu4-isys-csi2.c
index 2c87487..5292575 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-isys-csi2.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-isys-csi2.c
@@ -54,12 +54,6 @@ static void trigger_error(struct ipu_isys_csi2 *csi2)
 	spin_unlock_irqrestore(&csi2->isys->lock, flags);
 }
 
-void ipu_isys_csi2_trigger_error(struct ipu_isys_csi2 *csi2)
-{
-	csi2->isys->reset_needed = true;
-	trigger_error(csi2);
-}
-
 void ipu_isys_csi2_trigger_error_all(struct ipu_isys *isys)
 {
 	int i;
@@ -345,7 +339,7 @@ void ipu_isys_csi2_error(struct ipu_isys_csi2 *csi2)
 		dev_err_ratelimited(&csi2->isys->adev->dev,
 				"csi2-%i received fatal error\n",
 				csi2->index);
-		ipu_isys_csi2_trigger_error(csi2);
+		ipu_isys_csi2_trigger_error_all(csi2->isys);
 	}
 
 	for (i = 0; i < ARRAY_SIZE(errors); i++) {
-- 
2.7.4

